README.md
# bischool-theme
bi School Departments Child Theme


CHANGELOG.md
**Viterbi Sub-site Theme Change Log**

**5.0 - January 21, 2022**
* TH - Theme cleanup as documented here: https://edu.sharepoint.com/:w:/s/bi-it/ER5cQSVy_2xOpOmAPWOL-mQBJ023CXf1sHyuS3fsNPRJQA?e=J4XJfi

**4.3.8 - November 10, 2021**
* EH - Converted (/css/site-header.css) to (/stylesheets/partials/_site-header.scss) and merged SASS file to (/stylesheets/site.scss)
* EH - Removed (/css/site-header.css) and (/css/site-header.css.map)
* EH - Renamed theme to Viterbi Sub-site Theme from viterbischool to avoid confusion. Rewrote main (style.css) header to WP standardization.

**4.3.7 - November 8, 2021**
* Updated "Last Modified" text to include "Published on" date (lib/base-functions/display_last_modified.php) (assets/stylesheets/partials/_customizations.scss:158)

**4.3.6 - November 2, 2021**
* EH - Updated ACF json file for new Areas of Affiliation: CHEMs field's `Harnessing Data Science` to `Harnessing Data Science and Machine Learning` (/acf-json/group_5aa94aa3b8985.json)
* EH - Updated ACF json file for new Areas of Affiliation: CHEMs field's `Harnessing Data Science` to `Harnessing Data Science and Machine Learning` (/acf-json/group_5ce808c8d8992.json)

**4.3.5 - October 19, 2021**
* EH - Updated ACF json file for new Areas of Affiliation: CHEMs field's `Developing Advanced Manufacturing Methods and Materials` to `Developing Advanced Materials and Manufacturing Methods` (/acf-json/group_5aa94aa3b8985.json)
* EH - Updated ACF json file for new Areas of Affiliation: CHEMs field's `Developing Advanced Manufacturing Methods and Materials` to `Developing Advanced Materials and Manufacturing Methods` (/acf-json/group_5ce808c8d8992.json)

**4.3.4 - October 14, 2021**
* EH - Updated ACF json file for new Areas of Affiliation: CHEMs field and modified Research Center field (/acf-json/group_5aa94aa3b8985.json)
* EH - Updated ACF json file for new Areas of Affiliation: CHEMs field and modified Research Center field (/acf-json/group_5ce808c8d8992.json)

**4.3.3 - September 2021**
* EH - FIXED: Transition issue with last child in the top navigation. (\lib\extended-functions\vsoe-site-settings\functions\header-style.php)
* EH - FIXED: Fixed onclick issue with menu item with nested sub-menus. (\assets\js\vsoe.js)

**4.3.2**
* FIXED: Removed Portfolio CPT in /lib/base-functions/remove-portfolio-cpt.php that stopped working.
* UPDATE: Updated admin notice to change our software update hours from 8:00am to 10:00am to 9:00am to 11:00am.

**4.3.1.2**

**July 2021**
* UPDATE: Commented out conflicting CSS in the lib > Extended Functions > VSOE Site Settings > Functions > header-style.php file.

**4.3.1.1**

**July 2021**
* EH - UPDATE: Commented out JS function in vsoe.js that was opening outside links in a new tab. This is no longer supported by Chrome as of v88.
* EH - UPDATE: Uncommented JS in vsoe.js that was changed in previous update.

**4.3.1**

**July 2021**
* UPDATE: Removed the ordering function Miles created for the Video Tutorials CPT.
* UPDATE: Removed browser warning Javascript in vsoe.js.
* FIXED: Commented out Javascript in vsoe.js that was causing custom links to open pop ups.

**4.2.9**

**April 2021**
* NEW: Added function to remove 1536x1536 and 2048x2048 images sizes that were added to WP 5.3.0

**4.2.8**

**April 2021**
* NEW: Created CSS to hide various bullets throughout the site

**4.2.7**
* FIXED: Added custom Javascript to vsoe.js to fix primary links clicking issues.
* FIXED: Added custom CSS to site.css to fix the x-megamenu's child element style issues.

**4.2.5**
* FIXED: Added custom Javascript fix to resolve primary links clickthrough issues
* FIXED: Added custom CSS to site.css to add cursor: pointer to primary links
* FIXED: Added new page-faculty-directory.php file created by Dana to fix missing administration filter
* NEW: Additional CSS fixes for The Viterbi Exchange page 

**4.2.4**
* NEW: Created a page template for the Viterbi Exchange Page. Named the page template file to page-imagine.php.
* NEW: Created a php file for Question Detail page. Named the newly added single php file to single-question.php. 
* NEW: Created CSS file for both Viterbi Exchange Page template and Question Detail page. Added viterbi-exchange.css and single-question.css in /assets/css.
* NEW: Created a JS file for both Viterbi Exchange Page template and Question Detail page. Added the imagine.js in /assets/js.
* NEW: Added a JS Library for the filter feature. We are using Isotope. A JS Library that will help us create a cool filter feature. Created a lib file under assets folder. Added the minified Isotope js file in /assets/lib.
* NEW: Created a CPT for Question Posts. Registered the Question Post Type in /lib/extended-functions/vsoe-question-cpt and imported this directory in functions.php line 168-169.
* UPDATE: Enqueue viterbi-exchange.css, single-question.css and imagine.js in /lib/base-functions/scripts-styles.php line 41-51. Also, created an if-condition statement that these enqueque scripts and styles are being called only for the page-imagine.php page template.
* UPDATE: Add CSS fixes from Theme Options to /assets/css/site.css for the hover issue. Line 2976 - 3015

**4.2.2**

**November 2020**

* NEW: ACF Pro meta fields added to Tutorial and Forum editor screens for display order
* NEW: Custom ‘orderly’ query arg to use display order in archive loop
* NEW: Breadcrumbs for Tutorial and Forum CPT added to /lib/base-functions/x-breadcrumbs.php

**4.2.1**

**November 2020**

* NEW: ‘Forum’ CPT activation added to Site Settings

functions.php
<?php

// ===================================================================================================
// FUNCTIONS.PHP
// ---------------------------------------------------------------------------------------------------
// Overwrite or add your own custom functions to X-Theme in this file.
// ===================================================================================================
//
// ===================================================================================================
// TABLE OF CONTENTS
// ---------------------------------------------------------------------------------------------------
//   01. Global functions
//   02. Enque scripts and styles
//   03. Custom login
//   04. Custom widgets/sidebars
//   05. If posts disabled in Site Settings removes post functionality from site
//   06. If posts enabled activates custom widget for blog posts via a shortcode
//   07. If enabled in Site Settings, displays a right-hand sidebar on blog and archive pages
//   08. Displays "last modified" text after $content
//   09. Enables a secondary navigation menu that's not included in X-Theme
//   10. Disables extra miage sizes
//   11. Disabled commenting functionality site-wide
//   12. Customizes REST query for custom post types and custom meta data
//   13. Removes all standard WP dashboard widgets
//   14. Disables Portfolio CPT included with X-Theme
//   15. Adds CSS classes to <basename(path)ody> for greater CSS specificity. Adds .vsoe to
//       all posts/pages/archives. Also adds .cs-layout for Cornerstone layout chosen in Site
//       Options or .wp-layout for standard WordPress layout. Adds .single-[slug] or
//       .page-[slug] to posts and pages. Adds .breadcrumbs-on if X Customizer option is set
//       to display breadcrumbs.
//   16. Displays update alert in WP admin on Mondays
//   17. Gives editors admin rights to Redirection plugin
//   18. Enables 'Read More' links on automatic post excerpts
//   19. Adds sortable 'last modified' column in WP admin
//   20. 'vsoe_back_button()' function that can be called from post/page templates
//       to display a 'Back' button to navigate to referring URL
//   21. Customizes HTML markup of default X-Theme breadcrumbs for compatability
//       with Viterbi School child theme markup and styling
//   22. Displays 'Read More' link on manually created excerpts
//   23. Enables assigning 'sticky' attribute in post editor; if assigned, post
//       will always appear first in archive view
//   24. Enables execution of shortcodes in widgets
//   25. Modifies post meta to display only post date
//   26. Creates 3 user roles: Author Plus, Editor Plus, Former Staff/Student
//   27. Extend posts preview to 30 days
//   28. Checks for instance of Gravity Forms plugin. If enabled, prevents email
//       to admin account on new form creation, and allows optional display of
//       form labels.
//   29. Creates the "Site Settings" options page
//   30. Creates 'Courses' CPT with custom taxonomies and widget and template displays
//   31. Widget to query for events by specified department URL and display via Appearance > Widgets
//   32. Folder Sizes Dashboard Widget
//   33. Creates 'Tutorials' CPT with custom taxonomies and template displays
//   34. Creates 'Forum' CPT with custom taxonomies and template displays
//   35. Creates 'Question' CPT with custom taxonomies and template displays
// ---------------------------------------------------------------------------------------------------

// ===================================================================================================
//   01. Global functions
// ===================================================================================================

    global $vsoe_layout;
    global $vsoe_bc;
    global $vsoe_pb;
    global $vsoe_ss;
    global $vsoe_sec_nav;
    global $vsoe_search_type;
    global $vsoe_ai;
    global $vsoe_courses_cpt;
    global $vsoe_events_feed;
    global $vsoe_blog_sidebar;
    global $vsoe_last_updated;

    global $vsoe_facebook;
    global $vsoe_twitter;
    global $vsoe_instagram;
    global $vsoe_youtube;
    global $vsoe_vimeo;
    global $vsoe_google;
    global $vsoe_pinterest;
    global $vsoe_linkedin;

    $vsoe_layout = get_field('vsoe_site_layout', 'option');
    $vsoe_bc = get_field('vsoe_enable_breadcrumbs', 'option');
    $vsoe_pb = get_field('vsoe_enable_postsblog', 'option');
    $vsoe_comments = get_field('vsoe_enable_comments', 'option');
    $vsoe_comments = get_field('vsoe_enable_comments', 'option');
    $vsoe_tutorials_cpt = get_field('enable_tutorials_custom_post_type');
    $vsoe_ss = get_field('vsoe_enable_social_media_sharing_widget', 'option');
    $vsoe_sec_nav = get_field('vsoe_add_secondary_navigation', 'option');
    $vsoe_search_type = get_field('vsoe_search', 'option');
    $vsoe_ai = get_field('vsoe_enable_author_images', 'option');
    $vsoe_courses_cpt = get_field('vsoe_enable_courses_custom_post_type', 'option');
    $vsoe_tutorials_cpt = get_field('enable_tutorials_custom_post_type', 'option');
    $vsoe_questions_cpt = get_field('enable_questions_custom_post_type', 'option');
    $vsoe_forum_cpt = get_field('enable_forum_custom_post_type', 'option');
    $vsoe_events_feed = get_field('vsoe_enable_events_feed', 'option');
    $vsoe_blog_sidebar = get_field('vsoe_sidebar_on_blog_landing_page', 'option');

    $vsoe_facebook = get_field('vsoe_footer_social_facebook', 'option');
    $vsoe_twitter = get_field('vsoe_footer_social_twitter', 'option');
    $vsoe_instagram = get_field('vsoe_footer_social_instagram', 'option');
    $vsoe_youtube = get_field('vsoe_footer_social_youtube', 'option');
    $vsoe_vimeo = get_field('vsoe_footer_social_vimeo', 'option');
    $vsoe_google = get_field('vsoe_footer_social_google', 'option');
    $vsoe_pinterest = get_field('vsoe_footer_social_pinterest', 'option');
    $vsoe_linkedin = get_field('vsoe_footer_social_linkedin', 'option');
    $vsoe_last_updated = get_field('vsoe_display_last_updated_text', 'option');

// =================================================================================================
//   02. Enque scripts and styles
// =================================================================================================

    require_once( __DIR__ . '/lib/base-functions/scripts-styles.php');

// =================================================================================================
//   03. Custom login
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/custom-login.php');

// =================================================================================================
//   04. Custom widgets/sidebars
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/custom-widgets.php');

// =================================================================================================
//   05. If posts disabled in Site Settings removes post functionality from site
// =================================================================================================
    
    if ($vsoe_pb == '0') {
      require_once( __DIR__ . '/lib/base-functions/disable-posts.php');
    }
    
// =================================================================================================
//   06. If posts enabled activates custom widget for blog posts via a shortcode
// =================================================================================================

    if ($vsoe_pb == '1') {
        require_once( __DIR__ . '/lib/extended-functions/vsoe-blog-feed/vsoe-blog-feed.php');
    }

// =================================================================================================
//   07. If enabled in Site Settings, displays a right-hand sidebar on blog and archive pages
// =================================================================================================
    
    if ($vsoe_blog_sidebar == '1') {
      require_once( __DIR__ . '/lib/base-functions/blog-sidebar.php');
    }
    
// =================================================================================================
//   08. Displays "last modified" text after $content
// =================================================================================================
    
    if ($vsoe_last_updated == '1') {
        require_once( __DIR__ . '/lib/base-functions/display_last_modified.php');
    }

// =================================================================================================
//   09. Enables secondary navigation menu that's not included in X-Theme
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/register-nav-menus.php');

// =================================================================================================
//   10. Disables extra miage sizes
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/remove-extra-image-sizes.php');

// =================================================================================================
//   11. Disabled commenting functionality site-wide
// =================================================================================================
    
    if ($vsoe_comments == '0' ) {
        require_once( __DIR__ . '/lib/base-functions/disable-comments.php');
    }

// =================================================================================================
//   12. Customizes REST query for custom post types and custom meta data
// =================================================================================================

    require_once( __DIR__ . '/lib/base-functions/rest-query-vars.php');

// =================================================================================================
//   13. Removes all standard WP dashboard widgets
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/remove-dashboard-widgets.php');

// =================================================================================================
//   14. Disables Portfolio CPT included with X-Theme
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/remove-portfolio-cpt.php');

// =================================================================================================
//   15. Adds CSS classes to <basename(path)ody> for greater CSS specificity. Adds .vsoe to
//       all posts/pages/archives. Also adds .cs-layout for Cornerstone layout chosen in Site
//       Options or .wp-layout for standard WordPress layout. Adds .single-[slug] or
//       .page-[slug] to posts and pages. Adds .breadcrumbs-on if X Customizer option is set
//       to display breadcrumbs.
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/add-body-class.php');

// =================================================================================================
//   16. Displays update alert in WP admin on Mondays
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/admin-notice.php');

    
// =================================================================================================
//   17. Gives editors admin rights to Redirection plugin
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/redirection-role.php');

// =================================================================================================
//   18. Enables 'Read More' links on automatic post excerpts
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/excerpt-more.php');

// =================================================================================================
//   19. Adds sortable 'last modified' column in WP admin
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/admin-columns.php');

// =================================================================================================
//   20. 'vsoe_back_button()' function that can be called from post/page templates
//       to display a 'Back' button to navigate to referring URL
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/back-button.php');
    
// =================================================================================================
//   21. Customizes HTML markup of default X-Theme breadcrumbs for compatability
//       with Viterbi School child theme markup and styling
// =================================================================================================

    require_once( __DIR__ . '/lib/base-functions/x-breadcrumbs.php');

// =================================================================================================
//   22. Displays 'Read More' link on manually created excerpts
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/manual-excerpt-more.php');

// =================================================================================================
//   23. Enables assigning 'sticky' attribute in post editor; if assigned, post
//       will always appear first in archive view
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/sticky-archive-class.php');

// =================================================================================================
//   24. Enables execution of shortcodes in widgets
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/shortcode-in-widget.php');

// =================================================================================================
//   25. Modifies post meta to display only post date
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/custom-post-meta.php');
    
// =================================================================================================
//   26. Creates 3 user roles: Author Plus, Editor Plus, Former Staff/Student
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/user-roles.php');

// =================================================================================================
//   27. Extend posts preview to 30 days
// =================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/posts-preview.php');

// =================================================================================================
//   28. Checks for instance of Gravity Forms plugin. If enabled, prevents email
//       to admin account on new form creation, and allows optional display of
//       form labels.
// =================================================================================================
    
    if ( class_exists( 'GFCommon' ) ) {
        require_once( __DIR__ . '/lib/base-functions/disable-gforms-notification.php');
    }
	
// =================================================================================================
//   29. Creates the "Site Settings" options page
// =================================================================================================
    
    require_once( __DIR__ . '/lib/extended-functions/vsoe-site-settings/vsoe-site-options.php');

// =================================================================================================
//   30. Creates 'Courses' CPT with custom taxonomies and widget and template displays
// =================================================================================================
    
    if ($vsoe_courses_cpt == '1') {
        require_once( __DIR__ . '/lib/extended-functions/vsoe-courses-cpt/vsoe-courses-cpt.php');
    }

// ===================================================================================================
//   31. Widget to query for events by specified department URL and display via Appearance > Widgets
// ===================================================================================================
    
    require_once( __DIR__ . '/lib/extended-functions/vsoe-events-widget/vsoe-events-widget.php');

// ===================================================================================================
//   32. Folder Sizes Dashboard Widget
// ===================================================================================================
    
    require_once( __DIR__ . '/lib/base-functions/folder-sizes-dashboard-widget.php');
    
// ===================================================================================================
//   33. Creates 'Tutorials' CPT with custom taxonomies and template displays
// ===================================================================================================
    
    if ($vsoe_tutorials_cpt == '1') {
        require_once( __DIR__ . '/lib/extended-functions/tutorial-cpt/tutorial-cpt.php');
    }

// =================================================================================================
//   34. Creates 'Forum' CPT with custom taxonomies and template displays
// =================================================================================================
    
    if ($vsoe_forum_cpt == '1') {
        require_once( __DIR__ . '/lib/extended-functions/forum-cpt/forum-cpt.php');
    }
    
// =================================================================================================
//   35. Creates 'Question' CPT with custom taxonomies and template displays
// =================================================================================================

    if($vsoe_questions_cpt == '1') {
      require_once( __DIR__ . '/lib/extended-functions/vsoe-question-cpt/vsoe-question-cpt.php');
    }
